plugins/MauticCrmBundle/Api/PipedriveApi.php

```diff
@@ -230,6 +234,97 @@ private function getRequestParameters(array $data = [])
         ];
     }
 
+    /**
+     * @param array $data
+     *
+     * @return array
+     */
+    public function addActivity($data)
+    {
+        $params     = $this->getRequestParameters($data);
+        $url        = sprintf('%s/%s', $this->integration->getApiUrl(), self::ACTIVITIES);
+        $response   = $this->transport->post($url, $params);
+
+        return $data = $this->getResponseData($response);
+    }
+
+    /**
+     * @param array $event
+     *
+     * @return array|mixed
+     */
+    public function getActivityType($event)
```

Comment:
Plus I'd recommend to decide what type I need. It looks like you need rather `hasActivityType` that would return bool.

`getActivityType` should return an array (or even better some DTO instead of the array) only and throw an exception if not found. The reason is readability. This tells you that the activity type should be created because it was not found.
```
try {
    $this->getIntegration()->getApiHelper()->getActivityType($event));
} catch (ActivityTypeNotFoundException $e) {
    $this->getIntegration()->getApiHelper()->createActivityType($event);
}
```
In the original code you don't see the reason why it should be created.
```
if (!$this->getIntegration()->getApiHelper()->getActivityType($event)) {
    $this->getIntegration()->getApiHelper()->createActivityType($event);
}
```
Plus in the future when we will update the code for PHP 7+ with method return types it will be easy when each method returns only one type instead of mixed.

Reply:
I use refactor it to hasActivityType
I think It was just old code, I did plan use getActivityType, but now it's not needed.